debezium-core/src/main/java/io/debezium/relational/TableIdParser.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.debezium.text.ParsingException;
+import io.debezium.text.TokenStream;
+import io.debezium.text.TokenStream.CharacterStream;
+import io.debezium.text.TokenStream.Tokenizer;
+import io.debezium.text.TokenStream.Tokens;
+
+/**
+ * Parses identifiers into the corresponding parts of a {@link TableId}.
+ *
+ * @author Gunnar Morling.
+ */
+class TableIdParser {
+
+    private static final char SEPARATOR = '.';
+
+    public static List<String> parse(String identifier, char quotingChar) {
+        TokenStream stream = new TokenStream(identifier, new TableIdTokenizer(identifier, quotingChar), true);
+        stream.start();
+
+        List<String> parts = new ArrayList<>();
+
+        while(stream.hasNext()) {
+            parts.add(stream.consume());
+        }
+
+        return parts;
+    }
+
+    private static class TableIdTokenizer implements Tokenizer {
```

Comment:
Well, here's rough code that parses a name into segments (since that's what you did in the PR) using the `BasicTokenizer`:

```
  protected List<String> tokenize(String content) {
    if (content == null || content.trim().isEmpty()) return Collections.emptyList();
    tokens = new TokenStream(content, tokenizer, true);
    tokens.start();
    List<String> segments = new ArrayList<>();
    StringBuilder currentSegment = new StringBuilder();
    while (tokens.hasNext()) {
      if (tokens.canConsume(BasicTokenizer.DECIMAL)) {
        // We've hit a delimiter ...
        segments.add(currentSegment.toString());
        currentSegment.setLength(0);
        continue;
      } else if (tokens.matchesAnyOf(BasicTokenizer.SINGLE_QUOTED_STRING, BasicTokenizer.DOUBLE_QUOTED_STRING)) {
        // No matter what we take, add it to the current segment ...
        String quoted = tokens.consume();
        currentSegment.append(quoted.substring(1, quoted.length() - 1));
      } else {
        // No matter what we take, add it to the current segment ...
        currentSegment.append(tokens.consume());
      }
    }
    segments.add(currentSegment.toString());
    return segments;
  }
```
One reason this is a bit more verbose is that `BasicTokenizer` creates separate tokens for symbols, which depending upon how flexible we wanted to be we would treat as part of the name. Here's part of a unit test for this method:

```
    assertThat(tokenize("foo")).containsExactly("foo");
    assertThat(tokenize("foo..")).containsExactly("foo", "", "");
    assertThat(tokenize("foo.bar.baz")).containsExactly("foo", "bar", "baz");
    assertThat(tokenize("foo.\"bar\".baz")).containsExactly("foo", "bar", "baz");
    assertThat(tokenize("\"foo\".\"bar\".\"baz\"")).containsExactly("foo", "bar", "baz");
    assertThat(tokenize("\"foo\".\"\".\"baz\"")).containsExactly("foo", "", "baz");
    assertThat(tokenize("\"foo\".\"bar'other'\".\"baz\"")).containsExactly("foo", "bar'other'", "baz");
    assertThat(tokenize("foo!@#$%^&*().bar{}[];:,<>?/.baz")).containsExactly("foo!@#$%^&*()", "bar{}[];:,<>?/", "baz");
```

However, if you wanted to create a custom `Tokenizer`, you could essentially copy what's in `BasicTokenizer` and remove the [lines that deal with symbol characters](https://github.com/debezium/debezium/blob/master/debezium-core/src/main/java/io/debezium/text/TokenStream.java#L2424-L2445) and leave everything else. In this case, we know that a token is either an unquoted or quoted word, and only need to remove the quotes if they exist:

```
  protected List<String> tokenize(String content) {
    if (content == null || content.trim().isEmpty()) return Collections.emptyList();
    tokens = new TokenStream(content, tokenizer, true);
    tokens.start();
    List<String> segments = new ArrayList<>();
    while (tokens.hasNext()) {
      if (tokens.canConsume(BasicTokenizer.DECIMAL)) {
        // We've hit a delimiter, but throw it away ...
      } else if (tokens.matchesAnyOf(BasicTokenizer.SINGLE_QUOTED_STRING, BasicTokenizer.DOUBLE_QUOTED_STRING)) {
        String quoted = tokens.consume();
        segments.add(quoted.substring(1, quoted.length() - 1));
      } else {
        segments.add(tokens.consume());
      }
    }
    return segments;
  }
```

If you do this and create a `TableTokenizer` class, I suspect that the previous generic code could be cleaned up and made more efficient since we know that table identifiers have between 1 and 3 name segments:

```
  protected TableId parse(String content) {
    if (content == null || content.trim().isEmpty()) return Collections.emptyList();
    tokens = new TokenStream(content, tokenizer, true);
    tokens.start();
    String seg1 = consumeSegment(tokens);
    consumeDelimiter(tokens);
    String seg2 = consumeSegment(tokens);
    consumeDelimiter(tokens);
    String seg3 = consumeSegment(tokens);
    if (seg3 != null) return new TableId(seg1, seg2, seg3);
    if (seg2 != null) return new TableId(seg1, null, seg2);
    if (seg1 != null) return new TableId(null, null, seg1);
    return null;
  }
  
  protected boolean consumeDelimiter(TokenStream tokens) {
    return tokens.canConsume(TableTokenizer.DECIMAL);
  }
  
  protected String consumeSegment(TokenStream tokens) {
    if (tokens.hasNext()) {
      if (tokens.matchesAnyOf(TableTokenizer.SINGLE_QUOTED_STRING, BasicTokenizer.DOUBLE_QUOTED_STRING)) {
        // No matter what we take, add it to the current segment ...
        String quoted = tokens.consume();
        return quoted.substring(1, quoted.length() - 1);
      }
      return tokens.consume();
    }
    return null;
  }
```

Hope this helps.